server/build_without_gradle.sh

```diff
@@ -9,14 +9,24 @@
 #
 #     BUILD_DIR=my_build_dir ./build_without_gradle.sh
 
-set -e
+set -eo pipefail
 
 SCRCPY_DEBUG=false
 SCRCPY_VERSION_NAME=1.16
 
 PLATFORM=${ANDROID_PLATFORM:-29}
 BUILD_TOOLS=${ANDROID_BUILD_TOOLS:-29.0.2}
 
+if [ -z ${ANDROID_HOME+x} ]; then
+    AIDL="$ANDROID_HOME/build-tools/$BUILD_TOOLS/aidl"
+    DX="$ANDROID_HOME/build-tools/$BUILD_TOOLS/dx"
+else
+    [ -x "$(which dx)" ] && DX="dx" \
+        || (echo "Please provide \$ANDROID_HOME or set \$PATH to include 'dx' executable"; exit 1)
+    [ -x "$(which aidl)" ] && AIDL="aidl" \
+        || (echo "Please provide \$ANDROID_HOME or set \$PATH to include 'aidl' executable"; exit 1)
+fi
+
```

Comment:
For info, why do you need this? Where do your `dx` and `aidl` come from?

(That way, the `dx` and `aidl` version are not necessarily from the expected platform.)

Reply:
> For info, why do you need this? Where do your dx and aidl come from?

The problem I'm trying to solve here is building `scrcpy` completely from scratch with `android-tools` compiled from source. On Arch Linux `scrcpy` is built using a prebuilt `scrcpy-server`. 

Because AOSP source has a huge and monolithic build system and because there's no proper tagging, Arch Linux, to build everything from source, need to do this: https://github.com/archlinux/svntogit-community/blob/packages/android-tools/trunk/generate_build.rb#L3 .

> (That way, the dx and aidl version are not necessarily from the expected platform.)

You're right. In fact, I didn't covered this, if you depend on a specific `$PLATFORM` and `$BUILD_TOOLS` versions, then, this is a problem.